### PR TITLE
Improve group members view.

### PR DESCRIPTION
**Pull request by charlesmchen**
* Use `ContactTableViewCell`.
   * Show avatars
   * Style contact names as they appear elsewhere.
   * Show block status.
* Tapping on a member now offers the following options:
   * Contact info (previously the only option).
   * Block/unblock this user.
   * Send 1:1 message to this user.
   * Call this user.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/MessagesViewController.h [hunk 1]
```diff
@@ -41,7 +41,9 @@ extern NSString *const OWSMessagesViewControllerDidAppearNotification;
 @property (nonatomic, strong) AVAudioPlayer *audioPlayer;
 @property (nonatomic, strong) AVAudioRecorder *audioRecorder;
 
-- (void)configureForThread:(TSThread *)thread keyboardOnViewAppearing:(BOOL)keyboardAppearing;
+- (void)configureForThread:(TSThread *)thread
+    keyboardOnViewAppearing:(BOOL)keyboardAppearing
+        callOnViewAppearing:(BOOL)callOnViewAppearing;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MessagesViewController.h]: This isn't great, but I'm not going to change it here.

File: Signal/src/ViewControllers/ShowGroupMembersViewController.m [hunk 2]
```diff
@@ -62,11 +100,15 @@ - (void)viewDidLoad {
 
     self.groupContacts =
         [[GroupContactsResult alloc] initWithMembersId:self.thread.groupModel.groupMemberIds without:nil];
-
+    __weak ShowGroupMembersViewController *weakSelf = self;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ShowGroupMembersViewController.m]: I think I've fixed a serious memory leak that would prevent this view from every being deallocated.

File: Signal/src/Storyboard/Main.storyboard [hunk 3]
```diff
@@ -1,11 +1,11 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="11762" systemVersion="15G1217" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" colorMatched="YES" initialViewController="tuk-0x-yCb">
```
Comment (michaelkirk) [inline Signal/src/Storyboard/Main.storyboard]: Sorry that I'm crazy and read the storyboard diffs, but it seems like there aren't any meaningful changes here? Did you intend to check this in?

File: Signal/src/ViewControllers/ShowGroupMembersViewController.m [hunk 4]
```diff
@@ -90,44 +134,176 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
...
+    // Adjust index path for the header row.
+    indexPath = [NSIndexPath indexPathForRow:indexPath.row - 1 inSection:indexPath.section];
+
+    UIAlertController *actionSheetController =
+        [UIAlertController alertControllerWithTitle:nil message:nil preferredStyle:UIAlertControllerStyleActionSheet];
+
+    [actionSheetController
+        addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"GROUP_MEMBERS_VIEW_CONTACT_INFO",
+                                                     @"Button label for the 'show contact info' button")
+                                           style:UIAlertActionStyleDestructive
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ShowGroupMembersViewController.m]: None of these actions should be styled as "destructive" except "Block"

File: Signal/src/ViewControllers/ShowGroupMembersViewController.m [hunk 5]
```diff
@@ -154,15 +329,58 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
...
 - (Contact *)contactForIndexPath:(NSIndexPath *)indexPath {
     Contact *contact = [self.groupContacts contactForIndexPath:indexPath];
     return contact;
 }
 
+- (BOOL)isContactBlocked:(Contact *)contact
+{
+    if (contact.parsedPhoneNumbers.count < 1) {
+        // Hide contacts without any valid phone numbers.
+        return NO;
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ShowGroupMembersViewController.m]: I'm not sure what this comment means in this context. It doesn't seem like we're using this method to "hide" anything.

File: Signal/src/Storyboard/Main.storyboard (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/Storyboard/Main.storyboard]: Done.

File: Signal/src/ViewControllers/ShowGroupMembersViewController.m (hunk 4 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/ShowGroupMembersViewController.m]: Done.

File: Signal/src/ViewControllers/ShowGroupMembersViewController.m (hunk 5 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/ShowGroupMembersViewController.m]: Done.

**Pull request merged**